lib/parse.js

```diff
@@ -87,6 +87,10 @@ var parseValues = function parseQueryStringValues(str, options) {
             val = val.split(',');
         }
 
+        if(part.indexOf('[]=') > -1){
```

Comment:
``` suggestion
        if (part.indexOf('[]=') > -1) {
```
what happens with `foo[][]=bar`, or just `[]=a`?

Reply:
@ljharb 

`foo[][]=bar` will be parsed to ```{ foo: ['bar'] }```

`[]=a` will be parsed to  ```{ 0: 'a' }```

my code doesn't affect these results, it will be the same result